File: android/src/main/kotlin/com/onlinepayments/client/flutter/online_payments_sdk/validator/models/PaymentRequestValidationRequest.kt
/*
 * Do not remove or alter the notices in this preamble.
 *
 * This software is owned by Worldline and may not be be altered, copied, reproduced, republished, uploaded, posted, transmitted or distributed in any way, without the prior written consent of Worldline.
 *
 * Copyright © 2023 Worldline and/or its affiliates.
 *
 * All rights reserved. License grant and user rights and obligations according to the applicable license agreement.
 *
 * Please contact Worldline for questions regarding license and user rights.
 */

package com.onlinepayments.client.flutter.online_payments_sdk.validator.models

import com.onlinepayments.sdk.client.android.model.PaymentRequest
import com.onlinepayments.sdk.client.android.model.validation.ValidationErrorMessage

data class PaymentRequestValidationRequest(
    val paymentRequest: PaymentRequest,
) : ValidationRequest {
    override fun validate(): List<ValidationErrorMessage> {
        return paymentRequest.validate() ?: listOf(ValidationErrorMessage("Payment product or Payment Product Fields parsing failed and the field cannot be found on the payment request $this", paymentRequest.paymentProduct.id, null))
    }
}

File: lib/src/model/requests/validation/payment_request_validation_request.dart
/*
 * Do not remove or alter the notices in this preamble.
 *
 * This software is owned by Worldline and may not be be altered, copied, reproduced, republished, uploaded, posted, transmitted or distributed in any way, without the prior written consent of Worldline.
 *
 * Copyright © 2025 Worldline and/or its affiliates.
 *
 * All rights reserved. License grant and user rights and obligations according to the applicable license agreement.
 *
 * Please contact Worldline for questions regarding license and user rights.
 */
import 'package:json_annotation/json_annotation.dart';
import 'package:online_payments_sdk/online_payments_sdk.dart';

part 'payment_request_validation_request.g.dart';

/// Request used to validate a [PaymentProductField] in a [PaymentRequest].
@JsonSerializable(explicitToJson: true)
class PaymentRequestValidationRequest {
  @JsonKey(required: true)
  final PaymentRequest paymentRequest;

  @JsonKey(required: false)
  final String? fieldId;

  PaymentRequestValidationRequest(this.paymentRequest, {this.fieldId});

  factory PaymentRequestValidationRequest.fromJson(Map<String, dynamic> json) =>
      _$PaymentRequestValidationRequestFromJson(json);

  Map<String, dynamic> toJson() => _$PaymentRequestValidationRequestToJson(this);
}
